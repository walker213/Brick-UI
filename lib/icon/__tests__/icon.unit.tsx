import renderer from "react-test-renderer";
import React from "react";
import { mount } from 'enzyme';
import Icon from "../icon";

describe("Icon test", () => {
  it("render successfully", () => {
    const json = renderer.create(<Icon name='wechat' />).toJSON();
    expect(json).toMatchSnapshot();  // 如需更新最新snapshot，允许yarn test -u
	});
	it('onClick', () => {
		const fn = jest.fn();
		const component = mount(<Icon name='wechat' onClick={fn} />)
		component.find('svg').simulate('click')
		expect(fn).toBeCalled();
	})
});

