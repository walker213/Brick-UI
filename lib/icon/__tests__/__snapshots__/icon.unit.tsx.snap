// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon test render successfully 1`] = `
<svg
  className="bui-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
